### DISCUSSION TITLE
Can't setup SPI Device (SPI_CONFIG_DT)

### DISCUSSION THREAD

**Discussion by hufnag** [zephyrproject-rtos/zephyr#52957] [Q&A]
Hi all,
I'm currently struggling with SPI on a _nuclelo_u575zi_q_ board. To integrate the _NXP TJA1145A_ CAN-Transceiver into the device tree I wrote an  `app.overlay` . To use the `spi_transceive` and `spi_write` functions I need the `spi_config`. Therefore I use the `SPI_CONFIG_DT` macro. But during compilation, I get some weird errors that I can't understand.
e.g: 
```
/Users/user/zephyrproject/apps/tja-only/build/zephyr/include/generated/devicetree_generated.h:11909:34: error:
'DT_N_S_soc_S_spi_40013000_S_nxp_tja1145a_1_P_spi_max_frequency' undeclared here (not in a function);
did you mean 'DT_N_S_soc_S_spi_40013000_S_nxp_tja1145a_1_P_compatible_LEN'?
``` 
But the `spi-max-frequency` property is part of my overlay.
Do you have an idea what is going wrong?

Here the [source code](https://github.com/zephyrproject-rtos/zephyr/files/10205418/tja-spi.zip) and [stdout](https://github.com/zephyrproject-rtos/zephyr/files/10205419/compile_out.txt) from `west build -b nucleo_u575zi_q`. 
I'm using west@v0.14.0, zephyr@main(7b4fc1d6d) and zephyr-skd-0.15.1.
I appreciate any help you can provide!

**Comment by GeorgePriestner-Eaton** [comment]
Did you manage to resolve?

**Reply by ndrs-pst** [reply]
@GeorgePriestner-Eaton, I'm not sure if you're encountering the same problem as @hufnag since it's a Q&A from two years ago 🤔. 
At least, there is no DTS binding for such a device in the current Zephyr codebase. That's why the compilation resulted in an error.
